tests/test_block_record_fidelity.py

```diff
@@ -129,3 +129,28 @@ def wrap_call(expr: str, br: Any) -> str:
         return f"V:{ret}"
     except Exception as e:
         return f"E:{e}"
+
+
+# TODO: more thoroughly check these new functions which use self
+def test_calculate_sp_iters():
+    ssi: uint64 = uint64(100001 * 64 * 4)
+    rng = Random()
+    rng.seed(1337)
+    br = get_block_record(rng, ssi=ssi, spi=31)
+    res = br.sp_iters_impl(DEFAULT_CONSTANTS)
+    assert res is not None
+
+
+def test_calculate_ip_iters():
+    ssi: uint64 = uint64(100001 * 64 * 4)
+    sp_interval_iters = ssi // 32
+    ri = sp_interval_iters - 1
+    rng = Random()
+    rng.seed(1337)
+    br = get_block_record(rng, ssi=ssi, spi=31, ri=ri)
+    with raises(ValueError):
+        res = br.ip_iters_impl(DEFAULT_CONSTANTS)
+
+    br = get_block_record(rng, ssi=ssi, spi=13, ri=1)
+    res = br.ip_iters_impl(DEFAULT_CONSTANTS)
+    assert res is not None
```

Comment:
not `None`, it seems like the wrong value would also pass here

Reply:
Made this a specific value